### PR TITLE
feat: Implement List Repository Activities endpoint

### PR DESCRIPTION
**Pull request by AbhishekAg**
This PR implements support for the "List repository activities" REST API endpoint introduced in GitHub API v2022-11-28. This functionality allows go-github users to retrieve detailed activity information for repositories.

**Implementation Details**

- Added RepositoryActivity struct to represent activity data returned by the API
- Created ListRepositoryActivitiesOptions for pagination and filtering options (direction, before/after timestamps)
- Implemented ListActivities method in the RepositoriesService
- Added comprehensive tests for the new functionality
- Updated documentation with usage examples

**API Reference**
This implementation follows the GitHub REST API specification documented at:
https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-repository-activities
Example Usage
```
opts := &github.ListRepositoryActivitiesOptions{
    Direction: "desc",
    After:     "2023-01-01T00:00:00Z",
    ListOptions: github.ListOptions{
        Page:    1,
        PerPage: 30,
    },
}

activities, resp, err := client.Repositories.ListActivities(ctx, "owner", "repo", opts)
```
**Testing**

The implementation includes unit tests that verify:
- Correct URL formatting and parameter handling
- Proper deserialization of API responses
- Edge cases (empty results, error handling)

### CODE REVIEW

File: github/repos.go [hunk 1]
```diff
@@ -2438,3 +2438,75 @@ func (s *RepositoriesService) IsPrivateReportingEnabled(ctx context.Context, own
...
+type RepositoryActivityOptions struct {
+	Direction    string `url:"direction,omitempty"`
+	Before       string `url:"before,omitempty"`
+	After        string `url:"after,omitempty"`
+	Ref          string `url:"ref,omitempty"`
+	Actor        string `url:"actor,omitempty"`
+	TimePeriod   string `url:"time_period,omitempty"`
+	ActivityType string `url:"activity_type,omitempty"`
+
+	ListOptions
```
Comment (gmlewis) [inline github/repos.go]: Looking at the documentation: https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-repository-activities
this endpoint only uses `PerPage` from `ListOptions` and only `Before` and `After` from `ListCursorOptions`, which is a bit odd.

So I think you can completely remove `ListOptions` and add in `PerPage`.

Also, this whole struct needs a godoc-style comment, and it would be nice if each field listed its possible values.

File: github/repos.go [hunk 2]
```diff
@@ -2438,3 +2438,75 @@ func (s *RepositoriesService) IsPrivateReportingEnabled(ctx context.Context, own
...
+}
+
+// Actor represents a repository actor.
+type Actor struct {
```
Comment (gmlewis) [inline github/repos.go]: Let's please rename `Actor` to `RepositoryActor` since these names are package-global.

File: github/repos.go [hunk 3]
```diff
@@ -2438,3 +2438,75 @@ func (s *RepositoriesService) IsPrivateReportingEnabled(ctx context.Context, own
...
+	OrganizationsURL  *string `json:"organizations_url,omitempty"`
+	ReposURL          *string `json:"repos_url,omitempty"`
+	EventsURL         *string `json:"events_url,omitempty"`
+	ReceivedEventsURL *string `json:"received_events_url,omitempty"`
+	Type              *string `json:"type,omitempty"`
+	UserViewType      *string `json:"user_view_type,omitempty"`
+	SiteAdmin         *bool   `json:"site_admin,omitempty"`
+}
+
+type RepositoryActivity struct {
```
Comment (gmlewis) [inline github/repos.go:2501]: This struct needs a godoc-style comment.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @AbhishekAg.
Please address these items, then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

File: github/repos.go (hunk 1 above)
Comment (AbhishekAg) [reply github/repos.go]: Done

File: github/repos.go (hunk 3 above)
Comment (AbhishekAg) [reply github/repos.go:2501]: Done

File: github/repos.go (hunk 2 above)
Comment (AbhishekAg) [reply github/repos.go]: Done

File: openapi_operations.yaml
Comment (alexandear) [inline openapi_operations.yaml:1]: Could you clarify why this file was changed? From what I can see, the `GET /repos/{owner}/{repo}/activity` endpoint already exists in `master`:

https://github.com/google/go-github/blob/ac631dc3f83dcbfb6ffaffd8e7e28c59f2ad3264/openapi_operations.yaml#L3726-L3731

File: github/repos.go [hunk 4]
```diff
@@ -2438,3 +2438,96 @@ func (s *RepositoriesService) IsPrivateReportingEnabled(ctx context.Context, own
...
+	ActivityType string           `json:"activity_type"`
+	Actor        *RepositoryActor `json:"actor"`
+}
+
+// ListRespositoryActivities lists the activities for a repository.
+//
+// GitHub API docs: https://docs.github.com/rest/repos/repos#list-repository-activities
+//
+//meta:operation GET /repos/{owner}/{repo}/activity
+func (s *RepositoriesService) ListRespositoryActivities(ctx context.Context, owner, repo string, opts *RepositoryActivityOptions) ([]*RepositoryActivity, *Response, error) {
```
Comment (alexandear) [inline github/repos.go]: Please fix a typo:
```suggestion
func (s *RepositoriesService) ListRepositoryActivities(ctx context.Context, owner, repo string, opts *RepositoryActivityOptions) ([]*RepositoryActivity, *Response, error) {
```

File: github/repos.go [hunk 5]
```diff
@@ -2438,3 +2438,96 @@ func (s *RepositoriesService) IsPrivateReportingEnabled(ctx context.Context, own
 	resp, err := s.client.Do(ctx, req, privateReporting)
 	return privateReporting.Enabled, resp, err
 }
+
+// RepositoryActivityOptions specifies the optional parameters to the
+// RepositoriesService.ListRespositoryActivities method.
+type RepositoryActivityOptions struct {
+	// The direction to sort the results
```
Comment (alexandear) [inline github/repos.go]: Let's extend the description:
```suggestion
	// The direction to sort the results by. Can be one of: `asc`, `desc`. Default: `desc`. 
```

This can be found at GitHub docs:

image

Comment (alexandear) [reply github/repos.go]: The same comment applies to other fields.

File: github/repos.go [hunk 6]
```diff
@@ -2438,3 +2438,96 @@ func (s *RepositoriesService) IsPrivateReportingEnabled(ctx context.Context, own
...
+	// For paginated result sets, the number of results to include per page.
+	PerPage int `url:"per_page,omitempty"`
+
+	// A cursor, as given in the Link header. If specified, the query only searches for events before this cursor.
+	Before string `url:"before,omitempty"`
+
+	// A cursor, as given in the Link header. If specified, the query only searches for events after this cursor.
+	After string `url:"after,omitempty"`
+
+	// The Git reference for the activities you want to list.
```
Comment (alexandear) [inline github/repos.go:2459]:
```suggestion
	// The Git reference for the activities you want to list.
	// The ref for a branch can be formatted either as `refs/heads/<branch name>` or `<branch_name>`.
```

File: github/repos.go [hunk 7]
```diff
@@ -2438,3 +2438,96 @@ func (s *RepositoriesService) IsPrivateReportingEnabled(ctx context.Context, own
...
+// RepositoryActivity represents a repository activity.
+type RepositoryActivity struct {
+	ID           int64            `json:"id"`
+	NodeID       string           `json:"node_id"`
+	Before       string           `json:"before"`
+	After        string           `json:"after"`
+	Ref          string           `json:"ref"`
+	Timestamp    *Timestamp       `json:"timestamp"`
+	ActivityType string           `json:"activity_type"`
+	Actor        *RepositoryActor `json:"actor"`
```
Comment (alexandear) [inline github/repos.go]:
```suggestion
	Actor        *RepositoryActor `json:"actor,omitempty"`
```

Because:

```json
      "actor": {
        "anyOf": [
          {
            "type": "null"
          },
          {
            "title": "Simple User",
...
```

**Review by alexandear — CHANGES_REQUESTED** [review]

File: openapi_operations.yaml
Comment (AbhishekAg) [reply openapi_operations.yaml:1]: I agree it is already there, but I am not sure why it got generated. I ran `script/metadata.sh update-openapi` and I think this leads to generation.
I reverted these changes.

File: github/repos.go (hunk 6 above)
Comment (AbhishekAg) [reply github/repos.go:2459]: Taken care

File: github/repos.go (hunk 4 above)
Comment (AbhishekAg) [reply github/repos.go]: Thanks a lot for catching this. Corrected now

File: github/repos.go (hunk 5 above)
Comment (AbhishekAg) [reply github/repos.go]: Taken care

File: github/repos.go (hunk 7 above)
Comment (AbhishekAg) [reply github/repos.go]: Done

**Comment by AbhishekAg** [comment]
@alexandear Please review the PR when you get time.

File: github/repos.go [hunk 8]
```diff
@@ -2438,3 +2438,99 @@ func (s *RepositoriesService) IsPrivateReportingEnabled(ctx context.Context, own
 	resp, err := s.client.Do(ctx, req, privateReporting)
 	return privateReporting.Enabled, resp, err
 }
+
+// RepositoryActivityOptions specifies the optional parameters to the
+// RepositoriesService.ListRepositoryActivities method.
+type RepositoryActivityOptions struct {
```
Comment (alexandear) [inline github/repos.go]:
```suggestion
type ListRepositoryActivityOptions struct {
```

Comment (AbhishekAg) [reply github/repos.go]: Taken care

**Review by alexandear — APPROVED** [review]

**Comment by AbhishekAg** [comment]
@gmlewis @alexandear Can you please push these changes to the protected branch? I am not allowed to perform the push.

**Comment by gmlewis** [comment]
Thank you, @alexandear!
Merging.
